### PR TITLE
Make TryEnsureSufficientExecutionStack public

### PR DESCRIPTION
**Pull request by bartdesmet**
This addresses issue https://github.com/dotnet/corefx/issues/11090 per discussion over in the CoreFx repo.

### CODE REVIEW

File: src/mscorlib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs [hunk 1]
```diff
@@ -196,7 +196,7 @@ public static int OffsetToStringData
         [System.Security.SecuritySafeCritical]
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         [ReliabilityContract(Consistency.WillNotCorruptState, Cer.Success)]
-        internal static extern bool TryEnsureSufficientExecutionStack();
+        public static extern bool TryEnsureSufficientExecutionStack();
 #endif
```
Comment (weshaggard) [inline src/mscorlib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs]: Can you remove the FEATURE_CORECLR ifdef as well?

**Comment by weshaggard** [comment]
cc @jkotas

File: src/mscorlib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs [hunk 2]
```diff
@@ -188,16 +188,14 @@ public static int OffsetToStringData
...
         // This method ensures that there is sufficient stack to execute the average Framework function.
         // If there is not enough stack, then it return false.
         // Note: this method is not part of the CER support, and is not to be confused with ProbeForSufficientStack
         // below.
         [System.Security.SecuritySafeCritical]
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         [ReliabilityContract(Consistency.WillNotCorruptState, Cer.Success)]
-        internal static extern bool TryEnsureSufficientExecutionStack();
-#endif
+        public static extern bool TryEnsureSufficientExecutionStack();
```
Comment (jkotas) [inline src/mscorlib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs]: This is FCall - the ifdef needs to be removed in src/vm/ecalllist.h and src/vm/refllectioninvocation.* as well.

**Comment by jkotas** [comment]
This should be also added to src/mscorlib/model.xml. I believe that the corelib thinner won't mark it as public otherwise.

**Comment by bartdesmet** [comment]
Made changes to the places suggested in the review (and another one I've found by `findstr`); please let me know whether I covered them all now.

File: src/vm/reflectioninvocation.h [hunk 3]
```diff
@@ -56,10 +56,8 @@ class ReflectionInvocation {
 #endif // !FEATURE_CORECLR
     static FCDECL1(void, PrepareContractedDelegate, Object* delegateUNSAFE);
     static FCDECL0(void, ProbeForSufficientStack);    
-	static FCDECL0(void, EnsureSufficientExecutionStack);
-#ifdef FEATURE_CORECLR // currently only used from mscorlib in FEATURE_CORECLR
-	static FCDECL0(FC_BOOL_RET, TryEnsureSufficientExecutionStack);
-#endif // FEATURE_CORECLR
+    static FCDECL0(void, EnsureSufficientExecutionStack);
```
Comment (bartdesmet) [inline src/vm/reflectioninvocation.h]: There were some tabs here; converted to spaces.

**Comment by jkotas** [comment]
Thanks!

**Pull request merged**